File: LeamiServices/Database/Entities/RestaurantInfo.cs
using Leami.Model.Entities;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Leami.Services.Database.Entities
{
    public class RestaurantInfo
    {
        [Key]
        public int RestaurantId { get; set; }
        public string Name { get; set; } = string.Empty;
        public string? Description { get; set; } = string.Empty;
        public string Address { get; set; } = string.Empty;
        public string Phone { get; set; } = string.Empty;
        public byte[]? RestaurantImage { get; set; }
        public TimeSpan OpeningTime { get; set; }
        public TimeSpan ClosingTime { get; set; }
        public int? AdminUserId { get; set; }
        public virtual User? Admin { get; set; } = null!;

    }
}

File: Leami.UI/leami_desktop/lib/models/restaurant_info.dart
import 'package:json_annotation/json_annotation.dart';

part 'restaurant_info.g.dart';

@JsonSerializable()
class RestaurantInfo {
  int? restaurantId;
  String name;
  String? description;
  String? address;
  String? phone;
  String? restaurantImage;
  String openingTime;
  String closingTime; // "HH:mm:ss"

  RestaurantInfo({
     this.restaurantId,
    this.name = '',
    this.description,
    this.address,
    this.phone,
    this.restaurantImage,
    this.openingTime = '09:00:00',
    this.closingTime = '17:00:00',
  });

  factory RestaurantInfo.fromJson(Map<String, dynamic> json) =>
      _$RestaurantInfoFromJson(json);

  Map<String, dynamic> toJson() => _$RestaurantInfoToJson(this);
}
